### PR TITLE
Feat/cfe 206 proposal top

### PR DESCRIPTION
**Pull request by jennieramida**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -38,7 +38,7 @@
     "build-storybook": "storybook build"
   },
   "resolutions": {
-    "@interchain-ui/react": "1.11.4",
+    "@interchain-ui/react": "1.20.7",
```
Comment (songwongtp) [inline package.json:41]: revert

**Review by songwongtp — CHANGES_REQUESTED** [review]

File: package.json [hunk 2]
```diff
@@ -71,7 +71,7 @@
     "@graphql-typed-document-node/core": "^3.2.0",
     "@initia/initia.js": "0.1.12",
     "@initia/initia.proto": "0.1.12",
-    "@interchain-ui/react": "1.11.4",
+    "@interchain-ui/react": "1.20.7",
```
Comment (songwongtp) [inline package.json:74]: ditto

**Review by songwongtp — CHANGES_REQUESTED** [review]

File: package.json [hunk 3]
```diff
@@ -159,6 +159,7 @@
     "next-sitemap": "^3.1.25",
     "prettier": "^3.0.3",
     "storybook": "^7.6.9",
+    "storybook-react-context": "^0.6.0",
```
Comment (evilpeach) [inline package.json:162]: remove for now

File: src/lib/components/ExplorerLink.tsx [hunk 4]
```diff
@@ -75,13 +76,12 @@ export const getNavigationUrl = ({
...
     case "proposal_id":
-      url =
-        explorerConfig.proposal ||
-        `${lcdEndpoint}/cosmos/gov/v1beta1/proposals`;
+      url = "/proposals";
       break;
     case "pool_id":
       url = "/pools";
       break;
+
```
Comment (songwongtp) [inline src/lib/components/ExplorerLink.tsx:84]: remove empty line

File: src/lib/components/table/proposals/ProposalsTableMobileCard.tsx [hunk 5]
```diff
@@ -89,25 +90,13 @@ export const ProposalsTableMobileCard = ({
...
-            }
-          : undefined
-      }
+      onClick={() => {
+        trackMintScan("proposal-detail", {
+          types: proposal.types,
+          status: proposal.status,
+        });
+        onCardSelect(proposal.id);
+      }}
```
Comment (songwongtp) [inline src/lib/components/table/proposals/ProposalsTableMobileCard.tsx:99]: onClick={() => onCardSelect(proposal.id)}

**Review by songwongtp — CHANGES_REQUESTED** [review]

File: src/lib/services/proposal.ts [hunk 6]
```diff
@@ -180,14 +180,21 @@ const zProposalDataResponse = z.object({
...
-    .transform<ProposalData>(({ messages, ...val }) => ({
-      ...snakeToCamel(val),
-      messages,
-    }))
+    .transform<ProposalData>(
+      ({ created_tx_hash, proposal_deposits, messages, ...val }) => ({
+        ...snakeToCamel(val),
+        createdTxHash: created_tx_hash ? parseTxHash(created_tx_hash) : null,
+        proposalDeposits: proposal_deposits.map((deposit) => ({
+          ...deposit,
```
Comment (evilpeach) [inline src/lib/services/proposal.ts:192]: Let's address each property individually, given that the `deposit` includes a `tx_hash` property.

**Review by evilpeach — CHANGES_REQUESTED** [review]

**Review by songwongtp — CHANGES_REQUESTED** [review]

**Review by evilpeach — APPROVED** [review]

**Review by songwongtp — APPROVED** [review]

**Pull request merged**